lib/src/component/abstract_transition.dart

```diff
@@ -187,16 +190,22 @@ abstract class AbstractTransitionComponent<T extends AbstractTransitionProps, S
       skipCount = 0;
     }
 
-    var timer = new Timer(transitionTimeout, () {
+    _transitionEndTimer = new Timer(transitionTimeout, () {
       assert(ValidationUtil.warn(
         'The number of transitions expected to complete have not completed. Something is most likely wrong.'
       ));
 
+      _cancelTransitionEventListener();
       complete();
     });
 
     _endTransitionSubscription = getTransitionDomNode()?.onTransitionEnd?.skip(skipCount)?.take(1)?.listen((_) {
-      timer.cancel();
+
+      if (_transitionEndTimer != null) {
```

Comment:
There should be a private helper function to do this. Also you can just use a null-aware operator.

```dart
void _cancelTransitionEndTimer() {
  _transitionEndTimer?.cancel();
  _transitionEndTimer = null;
}
```